File: sdks/akta-docs/java/src/main/java/dev/openakta/aktadocs/RuleSeverity.java
package dev.openakta.aktadocs;

public final class RuleSeverity {
    private RuleSeverity() {}

    public static String effective(AktaConfig cfg, String ruleId) {
        RuleOptions ro = cfg.linter.rules.get(ruleId);
        if (ro != null && ro.severity != null) {
            return ro.severity;
        }
        return cfg.linter.defaultSeverity;
    }

    public static double numberOption(AktaConfig cfg, String ruleId, String key, double fallback) {
        RuleOptions ro = cfg.linter.rules.get(ruleId);
        if (ro == null) return fallback;
        return switch (key) {
            case "min_words" -> ro.minWords != null ? ro.minWords : fallback;
            case "max_words" -> ro.maxWords != null ? ro.maxWords : fallback;
            case "min_question_ratio" ->
                    ro.minQuestionRatio != null ? ro.minQuestionRatio : fallback;
            default -> fallback;
        };
    }

    public static String mapSeverity(String eff) {
        if ("off".equals(eff)) return null;
        return eff;
    }
}

File: sdks/akta-docs/csharp/RuleSeverity.cs
namespace OpenAkta.AktaDocs;

public static class RuleSeverity
{
    public static string Effective(AktaConfig cfg, string ruleId)
    {
        if (cfg.Linter.Rules.TryGetValue(ruleId, out var ro) && ro.Severity != null)
            return ro.Severity;
        return cfg.Linter.DefaultSeverity;
    }

    public static double NumberOption(AktaConfig cfg, string ruleId, string key, double fallback)
    {
        if (!cfg.Linter.Rules.TryGetValue(ruleId, out var ro) || ro == null) return fallback;
        return key switch
        {
            "min_words" => ro.MinWords ?? fallback,
            "max_words" => ro.MaxWords ?? fallback,
            "min_question_ratio" => ro.MinQuestionRatio ?? fallback,
            _ => fallback
        };
    }

    public static string? MapSeverity(string eff) => eff == "off" ? null : eff;
}
